sdk/identity/identity/src/msal/nodeFlows/msalClientAssertion.ts

```diff
@@ -33,20 +33,19 @@ export class MsalClientAssertion extends MsalNode {
 
   protected async doGetToken(
     scopes: string[],
-    options: CredentialFlowGetTokenOptions = {},
+    options: CredentialFlowGetTokenOptions = {}
   ): Promise<AccessToken> {
     try {
-      const assertion = await this.getAssertion();
       const result = await this.getApp(
         "confidential",
-        options.enableCae,
+        options.enableCae
       ).acquireTokenByClientCredential({
         scopes,
         correlationId: options.correlationId,
         azureRegion: this.azureRegion,
         authority: options.authority,
         claims: options.claims,
-        clientAssertion: assertion,
+        clientAssertion: this.getAssertion,
```

Comment:
👋 you probably already noticed this but OBO was migrated to MsalClient - this code has no effect (unless you revert to using MsalFlow again)

Instead, you could add it as a third overload here https://github.com/Azure/azure-sdk-for-js/blob/d131ca4d3f637e8c292207472133a79324b9fa42/sdk/identity/identity/src/msal/nodeFlows/msalClient.ts#L83-L104

You'll need to implement that overload here: https://github.com/Azure/azure-sdk-for-js/blob/d131ca4d3f637e8c292207472133a79324b9fa42/sdk/identity/identity/src/msal/nodeFlows/msalClient.ts#L714-L731

which means you will want to add a third conditional here https://github.com/Azure/azure-sdk-for-js/blob/d131ca4d3f637e8c292207472133a79324b9fa42/sdk/identity/identity/src/msal/nodeFlows/msalClient.ts#L734-L742

Reply:
Thanks Maor

Reply:
Did the required changes